chromium_src/components/content_settings/core/common/cookie_settings_base.cc

```diff
@@ -143,8 +144,49 @@ bool BraveIsAllowedThirdParty(
   return false;
 }
 
+// TODO(bridiver) make this a common method that can be shared with
+// BraveContentSettingsAgentImpl
+bool ShouldUseEphemeralStorage(
+    const GURL& url,
+    const GURL& site_for_cookies,
+    const base::Optional<url::Origin>& top_frame_origin,
+    const CookieSettingsBase* const cookie_settings) {
+  if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
+    return false;
+
+  if (!top_frame_origin || url::Origin::Create(url) == top_frame_origin)
+    return false;
+
+  bool block_3p = !cookie_settings->IsCookieAccessAllowed(
+      url, site_for_cookies, top_frame_origin);
+  bool block_1p = !cookie_settings->IsCookieAccessAllowed(
+      url, url, url::Origin::Create(url));
+
+  // only use ephemeral storage for block 3p
+  if (block_3p && !block_1p)
+    return true;
+
+  return false;
```

Comment:
Nit: Here you could simply do:
```c++
return block_3p && !block_1p;
```

Reply:
yea, I could still attach the same comment to it for clarity so I'll clean that up